frontend/src/services/HTTPHelper.ts

```diff
@@ -0,0 +1,368 @@
+
+      // Callback that is executed, when the load operation was aborted due to cancellation.
+      httpClient.onabort = () => {
+        // Unsubscribe from the cancellation token.
+        cancellationUnsubscribe();
+
+        reject(new CancellationError());
+      };
+
+      // Start the request
+      httpClient.send(JSON.stringify(body ?? {}));
+    });
+  }
+
+  public get<TResult>(
+    url: string | URL,
+    request: HTTPGETRequest,
+    cancellation?: CancellationToken
+  ): Promise<TResult>;
+
+  public get<TResult>(
+    url: string | URL,
+    cancellation?: CancellationToken
+  ): Promise<TResult>;
+
+  public async get<TResult>(
+    url: string | URL,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+    requestOrCancellation?: HTTPGETRequest | CancellationToken,
+    cancellation?: CancellationToken
+  ): Promise<TResult> {
+    let request: HTTPGETRequest;
+    let resolvedCancellation: CancellationToken | undefined = cancellation;
+    if (requestOrCancellation instanceof HTTPGETRequest) {
+      request = requestOrCancellation;
+    } else {
+      request = new HTTPGETRequest();
+      resolvedCancellation = requestOrCancellation;
+    }
+
+    const httpResponse = await this.tryGet<TResult>(
+      url,
+      request,
+      resolvedCancellation
+    );
+
+    // Check the HTTP response status code to be in the success range.
+    if (
+      httpResponse.status >= 200 &&
+      httpResponse.status <= 299 &&
+      httpResponse.result
+    ) {
+      return httpResponse.result;
+    }
+
+    throw Error(httpResponse.statusText);
+  }
+
+  public post<TResult>(
+    url: string | URL,
+    request: HTTPPOSTRequest,
+    cancellation?: CancellationToken
+  ): Promise<TResult>;
+
+  public post<TResult>(
+    url: string | URL,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+    body: any,
+    cancellation?: CancellationToken
+  ): Promise<TResult>;
+
+  public async post<TResult>(
+    url: string | URL,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+    requestOrBody: HTTPPOSTRequest | any,
+    cancellation?: CancellationToken
+  ): Promise<TResult> {
+    let request: HTTPPOSTRequest;
+    if (requestOrBody instanceof HTTPPOSTRequest) {
+      request = requestOrBody;
+    } else {
+      request = new HTTPPOSTRequest(requestOrBody);
+    }
+
+    const httpResponse = await this.tryPost<TResult>(
+      url,
+      request,
+      cancellation
+    );
+
+    // Check the HTTP response status code to be in the success range.
+    if (
+      httpResponse.status >= 200 &&
+      httpResponse.status <= 299 &&
+      httpResponse.result
+    ) {
+      return httpResponse.result;
+    }
+
+    throw Error(httpResponse.statusText);
+  }
+
+  private appendQuery(url: URL | string, query: URLQuery): URL {
```

Comment:
I'd move this method out of that file and make it only return a query string.
The URL creation with `this.options.baseUrl` is a side effect nobody expects from that function name.
Example:
```ts
createQueryParams(query: URLQuery): string {
     // ...
}

createQueryParams({x: 1, y: ['a', 'b']}) // returns x=1&y=a&y=b
```

Reply:
This function does also append this to a specified URL, which is non-trivial. But you are right, that the creation of the URL based on the baseUri is not the responsibility of this method.